### PR TITLE
[capi] add learning rate scheduler related api

### PR DESCRIPTION
**Pull request by lhs8928**
Commit 1:     [ccapi] rename LearningRateType enum name to LearningRateSchedulerType
    
     - rename enum name LearningRateType to LearningRateSchedulerType for more detailed info
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 2:     [ccapi] change setLearningRateScheduler function prototype
    
     - Change return type from void to int.
       Capi will call this function so it should be return status.
     - Change learning rate scheduler pointer from unique_ptr to shared_ptr
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 3:     [capi] add learning rate scheduler related api
    
     - Added learning rate scheduler create/destroy/set property api
     - Added set learning rate scheduler to optimizer
     - Added ml_train_lr_scheduler_type_e enum
     - Fix some comments
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 4:     [test] reorder tizen capi unittest
    
     - Reorder unittest for sequential order
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 5:     [capi] add unittest for learning rate scheduler
    
     - Added learning rate scheduler related unittest
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: api/ccapi/include/optimizer.h [hunk 1]
```diff
@@ -83,8 +83,8 @@ class Optimizer {
    *
    * @param lrs the learning rate scheduler object
    */
-  virtual void setLearningRateScheduler(
-    std::unique_ptr<ml::train::LearningRateScheduler> &&lrs) = 0;
+  virtual int setLearningRateScheduler(
+    std::shared_ptr<ml::train::LearningRateScheduler> lrs) = 0;
```
Comment (jijoongmoon) [inline api/ccapi/include/optimizer.h:87]: is there any reason you change the unique_ptr to shared_ptr?

File: test/tizen_capi/unittest_tizen_capi_lr_scheduler.cpp [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+ * @brief Learning rate scheduler set Property Test (positive test)
+ */
+TEST(nntrainer_capi_lr_scheduler, setProperty_01_p) {
+  ml_train_lr_scheduler_h handle;
+  int status;
+  status =
+    ml_train_lr_scheduler_create(&handle, ML_TRAIN_LR_SCHEDULER_TYPE_CONSTANT);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status =
+    ml_train_lr_scheduler_set_property(handle, "learning_rate=0.001", NULL);
```
Comment (jijoongmoon) [inline test/tizen_capi/unittest_tizen_capi_lr_scheduler.cpp:164]: we need to check if the user sets this scheduler property and model property differently.

Comment (lhs8928) [reply test/tizen_capi/unittest_tizen_capi_lr_scheduler.cpp:164]: We need to double check it but I guess in that case the optimizer will throw an error. (please refer https://github.com/nnstreamer/nntrainer/blob/main/nntrainer/optimizers/optimizer_wrapped.cpp#L96)

File: api/ccapi/include/optimizer.h (hunk 1 above)
Comment (lhs8928) [reply api/ccapi/include/optimizer.h:87]: Discussed offilne

**Review by jijoongmoon — APPROVED** [review]
LGTM
